lib/util/findNextPunctuators.js

```diff
@@ -44,87 +43,52 @@ module.exports = function(node, source, lines) {
     return punctuators;
   }
 
-  function collectVariableDeclarations(node, source) {
-    function isNewVariable(expression) {
-      return expression.type === 'Keyword' && -1 !== ['var', 'let', 'const'].indexOf(expression.value);
+  function getNextExpressions(source, node) {
+    function findNextExpressionToken(source, node) {
+      var isVarAssignment = isVariableAssignment(node), amount = !isVarAssignment ? 2 : 1;
+      return source.getTokenAfter(!isVarAssignment ? node.parent : node, amount);
     }
-
-    var isAssignment = isVariableAssignment(node),
-      after = source.getTokenAfter(node.declarations ? node : node.parent, isAssignment ? 0 : 1),
-      punctuators = getPunctuators(node, source);
-
-    // check the next line for further expressions and skip completely in case of empty lines
-    if (typeof after !== 'undefined') {
-      // checks whether a new variable is introduced
-      // or an existing one will be changed.
-      var punctuator = isNewVariable(after) ? source.getTokenAfter(after, 1) : source.getTokenAfter(after);
-
-      if (isPunctuatorAssignment(punctuator)) {
-        punctuators.push(punctuator);
+    function findProperPunctuator(next, x) {
+      var after = source.getTokenAfter(next, 2 - x);
+      if (after.value !== '=') {
+        return findProperPunctuator(after, 1);
```

Comment:
the recursion may cause issues if the second line doesn't have any vars declared